client/js/widgets/button.js

```diff
@@ -28,6 +28,27 @@ export class Button extends Widget {
       this.domElement.textContent = delta.text;
   }
 
+  applyVariables(field, widgets, variables, problems) {
```

Comment:
Isn't widgets a global variable that we are not modifying? Should not need to pass it in the function call but I also don't see harm.

Reply:
removed widgets from the function parameters.